android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java

```diff
@@ -1602,6 +1623,70 @@ public void onClick(View v) {
         }
     }
 
+    private void showVbatExpireNotice() {
+        if (mRewardsVbatExpireNoticeModal == null) {
+            return;
+        }
+
+        panelShadow(true);
+
+        mRewardsVbatExpireNoticeModal.setVisibility(View.VISIBLE);
+        TextView vBatModalTitle = mRewardsVbatExpireNoticeModal.findViewById(R.id.vbat_modal_title);
+        TextView vBatModalText = mRewardsVbatExpireNoticeModal.findViewById(R.id.vbat_modal_text);
+        FrameLayout vBatConnectButton =
+                mRewardsVbatExpireNoticeModal.findViewById(R.id.btn_vbat_connect_account);
+        vBatConnectButton.setVisibility(
+                mBraveRewardsNativeWorker.canConnectAccount() ? View.VISIBLE : View.GONE);
+        vBatConnectButton.setOnClickListener((new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                TabUtils.openUrlInNewTab(
+                        false, BraveActivity.BRAVE_REWARDS_SETTINGS_WALLET_VERIFICATION_URL);
+                dismiss();
+            }
+        }));
+        Button vBatLearnMoreButton =
+                mRewardsVbatExpireNoticeModal.findViewById(R.id.btn_vbat_learn_more);
+        vBatLearnMoreButton.setOnClickListener((new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                CustomTabActivity.showInfoPage(mActivity, BRAVE_REWARDS_CHANGES_PAGE);
+            }
+        }));
+        AppCompatImageView vbatCloseBtn =
+                mRewardsVbatExpireNoticeModal.findViewById(R.id.vbat_modal_close);
+        vbatCloseBtn.setOnClickListener((new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mRewardsVbatExpireNoticeModal != null) {
+                    mRewardsVbatExpireNoticeModal.setVisibility(View.GONE);
+                }
+                panelShadow(false);
+            }
+        }));
+
+        String vbatModalTitleString = "";
+        String vbatModalTextString = "";
+        double dueDateInMillis = mBraveRewardsNativeWorker.getVbatDeadline();
+        SimpleDateFormat sdf =
+                new SimpleDateFormat("MMMM dd, yyyy 'at' h:mm a", Locale.getDefault());
+        String dueDate = sdf.format(new Date((long) dueDateInMillis))
+                                 .replace("a.m.", "AM")
+                                 .replace("p.m.", "PM");
```

Comment:
AM/PM doesn't need to be localized?

Reply:
Do we localized those in desktop ? i am not sure.

Reply:
I don't recall ever having to worry about it, to be honest. It just sort of stuck out to me with the calls to `replace`. Might be worth double-checking. Would there even be an "a.m." or "p.m." to replace in non-English locales?

Reply:
tbh we shouldn't do this. we should use the locale default format.

Reply:
Will the `'at'` be localized correctly, or will that just be inserted in the resulting string? If it's not localized, then we'll need to use a different way to format the date, I suppose.

Reply:
@zenparsing in that case, we should use default android date formatting.